usecases/monitoring/grpc.go

```diff
@@ -40,13 +40,13 @@ const (
 //
 // ```
 //
-//	svrMetrics := monitoring.NewServerMetrics(metrics, prometheus.DefaultRegisterer)
+//	svrMetrics := monitoring.NewGRPCServerMetrics(metrics, prometheus.DefaultRegisterer)
 //	grpcServer := grpc.NewServer(monitoring.InstrumentGrpc(*svrMetrics)...)
 //
 //	grpcServer.Serve(listener)
 //
 // ```
-func InstrumentGrpc(svrMetrics ServerMetrics) []grpc.ServerOption {
+func InstrumentGrpc(svrMetrics *ServerMetrics) []grpc.ServerOption {
```

Comment:
hmm, I see in this method we are reusing HTTP server metrics for grpc. E.g. even the description is saying:
```go
RequestDuration: r.NewHistogramVec(prometheus.HistogramOpts{
	Namespace: namespace,
	Name:      "request_duration_seconds",
	Help:      "Time (in seconds) spent serving HTTP requests.",
	Buckets:   LatencyBuckets,
}, []string{"method", "route", "status_code"}),
```

I guess the idea is to have unified metrics of traffic, however, HTTP and gRPC status codes do not map one-to-one, so it might be confusing in the future, like you will have to not forget and write promql queries like this: `weaviate_request_duration_seconds_sum{instance="node-1", method="gRPC", route="/grpc.health.v1.Health/Check", status_code=~"200|OK"}`